docs/populators.md

```diff
@@ -0,0 +1,171 @@
+from pystac.extensions.file import FileExtension
+
+from STACpopulator.extensions.base import ExtensionHelper
+
+
+class FileHelper(ExtensionHelper):
+    access_urls: dict[str, str]
+    ...
+    session: Optional[Session] = Field(default=None, exclude=True)
+
+    @classmethod
+    def from_data(
+        cls,
+        data: dict[str, Any],
+        **kwargs,
+    ) -> "FileHelper":
+        """Create a FileHelper instance from raw data."""
+        return cls(access_urls=data["data"]["access_urls"], **kwargs)
+
+    def apply(self, item: pystac.Item, add_if_missing: bool = True) -> T:
+        asset = item.assets[...]
+        file_ext = FileExtension.ext(asset, add_if_missing=add_if_missing)
+        file_ext.apply(
+            size=self.size,
+            ...
+        )
+
+    @functools.cached_property
+    def size(self) -> Optional[int]:
+        """Return file size in bytes, None if an error occurs."""
+        size: int = ... # Code logic to calculate the file size
+        return size
+```
+
+## 3. Data Model
+
+Inheriting from [`BaseSTAC`](../STACpopulator/extensions/base.py#L159), a `DataModel` is the main data management class associated with an implementation in `stac-populator`. It holds `helper` objects as attributes, which it uses to apply multiple extensions to a given entity (i.e., `item`, `asset`, `link`, or `collection`).
+
+In `stac-populator` the [`THREDDSCatalogDataModel`](../STACpopulator/extensions/thredds.py#L173) class defines the minimal data model to preprocess and integrate data properties from a THREDDS Catalog. This catalog data model class includes the [`THREDDSHelper`](../STACpopulator/extensions/thredds.py#L133) used to add generic thredds properties from the catalog (e.g., `links`, `services`), and the [`DataCubeHelper`](../STACpopulator/extensions/datacube.py#L13) to add data cube-related properties (e.g., `dimensions`, `variables`, `bounds`).
+
+**A custom THREDDS data model class** must extend the `THREDDSCatalogDataModel` and only needs to define the additional helpers it should use. 
+
+Data model instances should be created using the `def from_data(...)` factory method, which accepts extra keyword arguments that are forwarded to the helper constructors. During instantiation, the data model automatically initializes its `helper` attributes by calling each helper’s own `def from_data(...)` method. The keyword arguments are distributed among the helpers based on their constructor signatures, allowing the data model to accept all required parameters in one place and dispatch them efficiently. The instantiated helpers are later automatically applied to extend the data model before integrating the data properties.
+
+
+The example below shows the `RDPSDataModel`. In addition to the default `THREDDSHelper` and `DataCubeHelper` inherited from `THREDDSCatalogDataModel`, this data model defines the `CFHelper` and the `FileHelper`, which respectively apply the `CFExtension` and `FileExtension`.
```

Comment:
Something should be mentioned regarding the attribute names (ie: `cf` and `file`).
We make it a "best-practice" for the helpers to use the same prefix as the STAC extension they handle, although this is not a requirement if a helper was doing something else.